project_euler/problem_72/sol1.py

```diff
@@ -0,0 +1,43 @@
+"""
+Consider the fraction, n/d, where n and d are positive integers. If n<d and HCF(n,d)=1,
+it is called a reduced proper fraction.
+
+If we list the set of reduced proper fractions for d ≤ 8 in ascending order of size,
+we get:
+
+1/8, 1/7, 1/6, 1/5, 1/4, 2/7, 1/3, 3/8, 2/5, 3/7, 1/2,
+4/7, 3/5, 5/8, 2/3, 5/7, 3/4, 4/5, 5/6, 6/7, 7/8
+
+It can be seen that there are 21 elements in this set.
+
+How many elements would be contained in the set of reduced proper fractions
+for d ≤ 1,000,000?
+"""
+
+
+def solution(limit: int = 1000000):
+    """
+    Return the number of reduced proper fractions with denominator less than limit.
+    >>> solution(8)
+    21
+    >>> solution(1000)
+    304191
+    """
+    primes = set(range(3, limit, 2))
+    primes.add(2)
+    for p in range(3, limit, 2):
+        if p not in primes:
+            continue
```

Comment:
```suggestion
```
These two lines are not needed.

Reply:
This is an optimisation - there's no point in removing the factors of p from primes if p isn't prime, as all its factors will already have been eliminated. 

Reference: [Sieve of Eratosthenes](https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes#Pseudocode)

    algorithm Sieve of Eratosthenes is
        let A be an array of Boolean values, indexed by integers 2 to n,
        initially all set to true.
        
        for i = 2, 3, 4, ..., not exceeding √n do
            if A[i] is true           << here
                for j = i2, i2+i, i2+2i, i2+3i, ..., not exceeding n do
                    A[j] := false
    
        return all i such that A[i] is true.

[Here's to a script I wrote that defines functions to generate primes with and without the optimisation, showing a speedup in that section of the code by about 50%.](https://pastebin.com/a1XQCjuk)

Reply:
Ah, I'm sorry I totally missed the part where you were updating the `set`.